examples/kubernetes/static_provisioning/secret_authentication.yaml

```diff
@@ -0,0 +1,89 @@
+# Secret Authentication Example
+# This example demonstrates using a Kubernetes Secret to provide S3 credentials for the Mountpoint S3 CSI Driver.
+# This authentication method is particularly useful for:
+# 1. The user wants to set their own credentials which are different than the driver level credentials
+# 2. Using different credentials for different persistent volumes
+
+# First, create a Secret containing the S3 credentials
+apiVersion: v1
+kind: Secret
+metadata:
+  name: s3-credentials
+  namespace: default
+type: Opaque
+data:
+  # Using base64 encoded values. Example:
+  # echo -n "ACCESS_KEY_ID" | base64
+  key_id: QUtJQVhYWFhYWFhYWFhYWFhY 
+  # echo -n "SECRET_ACCESS_KEY" | base64
+  access_key: U0VDUkVUWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWA==
+  
+  # You can also create the secret using kubectl:
+  # kubectl create secret generic s3-credentials \
+  #     --from-literal=key_id="ACCESS_KEY_ID" \
+  #     --from-literal=access_key="SECRET_ACCESS_KEY"
+  
+  # SECURITY CONSIDERATIONS:
+  # - Kubernetes Secrets are stored unencrypted in etcd by default
+  # - Consider enabling encryption at rest: https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/
```

Comment:
Could you enforce this by default or run a check to make sure the secrets are encrypted ?

Reply:
I am not sure if I understand, kubernetes secrets are secure.

Reply:
Actually I will remove this, we want to store StringData

Reply:
I removed this for now.
We use the same way to store Data and credentials, which will be kept secure.
[Initially, we were using plain values in volume definition](https://scality.atlassian.net/browse/S3CSI-5?focusedCommentId=412137), so it was not secure; it was a leftover comment.